### PR TITLE
Feature/export to native calendar

### LINKED ISSUE #783: View my schedule on my native calendar
**Is your feature request related to a problem? Please describe.**
As a student, I would like to have my schedules into the my native calendar.

### PR DESCRIPTION
**Pull request by mataai**
### ⁉️ Related Issue
Closes #783 

## 📖 Description
Implements a utility to interact in a very basic way with the android or iOS native calendar. Then uses it to export the course activities to the native calendar by allowing the user to choose to which calendar to send it to, or create a new one. Also supports updating of the even through hash code embedded in the even description. 

### 🧪 How Has This Been Tested?
Manual trial and error only.

### ☑️ Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [x] Make sure to add either one of the following labels: `version: Major`,`version: Minor` or `version: Patch`. 
- [x] Make sure golden files changes were reviewed and approved.

### 🖼️ Screenshots (if useful):
image
image
image

### CODE REVIEW

File: lib/core/utils/calendar_utils.dart [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      await deviceCalendarPlugin.createCalendar(calendarName);
+      calendar = await fetchNativeCalendar(calendarName);
+    }
+
+    // Fetch events from calendar to avoid duplicates
+    final events = await fetchNativeCalendarEvents(
+        calendar.id,
+        RetrieveEventsParams(
+          startDate: DateTime.now().subtract(const Duration(days: 120)),
+          endDate: DateTime.now().add(const Duration(days: 120)),
```
Comment (XavierPaquet-Rapold) [inline lib/core/utils/calendar_utils.dart:87]: Regarder si on peut exporter du début d'une session à la fin d'une session

**Review by XavierPaquet-Rapold — DISMISSED** [review]
LGTM 🔥

**Review by XavierPaquet-Rapold — DISMISSED** [review]
LGTM 🔥

**Review by Udito3 — COMMENTED** [review]
T'as pas fait de tests? :o

File: lib/core/utils/calendar_utils.dart [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    // Add events to calendar
+    for (final course in courses) {
+      // create event
+      final event = Event(
+        calendar.id,
+        title: course.courseName,
+        start: TZDateTime.from(
+            course.startDateTime, getLocation('America/Toronto')),
+        end:
+            TZDateTime.from(course.endDateTime, getLocation('America/Toronto')),
```
[the remark this answers is not in the document]
Comment (mataai) [reply lib/core/utils/calendar_utils.dart:112]: Perso j'ai utilisé la feature pour mon horraire de cette session et c'est la bonne heure. America/Toronto c'est le EST timezone pour le Canada.

**Review by XavierPaquet-Rapold — APPROVED** [review]

**Review by XavierPaquet-Rapold — DISMISSED** [review]

**Review by XavierPaquet-Rapold — DISMISSED** [review]

**Review by MysticFragilist — APPROVED** [review]
LGTM! 💯

**Review by JonathanDuvalV — APPROVED** [review]
Bien joué

**Review by Udito3 — APPROVED** [review]

**Pull request merged**